frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/quickfix/CreateBranchQuickFix.java

```diff
@@ -0,0 +1,66 @@
+package com.virtuslab.gitmachete.frontend.file.quickfix;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+
+import java.util.Collections;
+
+import com.intellij.codeInsight.intention.IntentionAction;
+import com.intellij.codeInspection.util.IntentionFamilyName;
+import com.intellij.codeInspection.util.IntentionName;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import git4idea.branch.GitBrancher;
+import git4idea.repo.GitRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.ExtensionMethod;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.IgnoreUIThreadUnsafeCalls;
+
+@RequiredArgsConstructor
+@ExtensionMethod({GitMacheteBundle.class})
+public class CreateBranchQuickFix implements IntentionAction {
+
+  private final String branch;
+  private final String parentBranch;
+  private final PsiFile macheteFile;
+  private final @Nullable GitRepository gitRepository;
+
+  @Override
+  public @IntentionName String getText() {
+    return getNonHtmlString("action.GitMachete.MacheteAnnotator.IntentionAction.create-nonexistent-branch")
+        .format(branch, parentBranch);
+  }
+
+  @Override
+  public @IntentionFamilyName String getFamilyName() {
+    return "Git Machete";
+  }
+
+  @Override
+  public boolean isAvailable(Project project, Editor editor, PsiFile file) {
+    return true;
+  }
+
+  @Override
+  public void invoke(Project project, Editor editor, PsiFile file) {
+    createNewBranchFromParent(project);
+  }
+
+  @Override
+  public boolean startInWriteAction() {
+    return false;
+  }
+
+  @IgnoreUIThreadUnsafeCalls
+  private void createNewBranchFromParent(Project project) {
+    if (gitRepository != null) {
+      GitBrancher.getInstance(project).createBranch(branch, Collections.singletonMap(gitRepository, parentBranch));
```

Comment:
Actually, AFAICS `createBranch` here ends up calling `git4idea.branch.GitBrancherImpl#createBranch(java.lang.String, java.util.Map<git4idea.repo.GitRepository,java.lang.String>, boolean)`, which itself just schedules a background task.

So it can be considered safe for running on UI thread. Pls remove `@IgnoreUIThreadUnsafeCalls` and instead patch `com.virtuslab.archunit.UIThreadUnsafeMethodInvocationsTestSuite#whitelistedMethodFullNames_git4idea` ✍🏻

Reply:
Huh, I tried to do that, but looks like my solution does not work.

Reply:
See below 👇🏻